src/server/game/Entities/Player/Player.cpp

```diff
@@ -23968,8 +23968,11 @@ void Player::SetViewpoint(WorldObject* target, bool apply)
         // farsight dynobj or puppet may be very far away
         UpdateVisibilityOf(target);
 
-        if (target->isType(TYPEMASK_UNIT) && target != GetVehicleBase())
+        if (target->isType(TYPEMASK_SEER) && target != GetVehicleBase())
```

Comment:
If adjusting `TYPEMASK_SEER` is not an option, then I will change the line being discussed to say "unit or player".